src/mail.sh

```diff
@@ -63,10 +67,42 @@ function validate_encryption()
   fi
 }
 
+# This function checks that the necessary options to use
+# the git send-email have been set
+function mail_verify()
+{
+  local count=0
+  local -a missing_conf
+  local set_confs
+
+  set_confs=$(git config --get-regexp 'user|sendemail\.smtp')
+  for config in "${essential_config_options[@]}"; do
+    if ! printf '%s\n' "$set_confs" | grep -cF "$config" - &> /dev/null; then
```

Comment:
Can we move this `printf` outside this `if`?

Reply:
The `printf` is just sending the value of `set_confs` to the `grep` function. This is testing if they are set, I think I will move them outside and capture the output to check the current value that is set.